### ISSUE TITLE
Upgrade to the latest version of d3 5.x

### ISSUE THREAD

**Issue by alexcojocaru** [nrenner/brouter-web#343]
Using an older version of d3 (i.e. 3.x) in the brouter-web app makes it difficult to integrate leaflet plugins which depend on the latest d3 (i.e. 5.x atm). When trying to integrate the [Leaflet.Heightgraph](https://github.com/GIScience/Leaflet.Heightgraph), as a replacement for the Leaflet.Elevation currently used to render the elevation profile of the route, and since Heightgraph depends on d3 5.x, I only had two options:

1. Backport the Heightgraph to d3 3.x, which I have done [without too much trouble](https://github.com/GIScience/Leaflet.Heightgraph/compare/master...alexcojocaru:d3-v3_2); going forward with this option means that:
   1. someone would have to maintain the fork and release it whenever needed (https://github.com/GIScience/Leaflet.Heightgraph/issues/101)
   1. as more and more d3 5.x APIs, with no direct translation to d3 3.x APIs, are used by the Heightgraph (e.g. [the color category](https://github.com/GIScience/Leaflet.Heightgraph/compare/master...alexcojocaru:d3-v3_2#diff-c50ee78a6e8e50ed26533866c2055c843c0fbaa273a90baf8399b96e217946f5R330)), the backporting could get complex and the fork could diverge significantly
1. Upgrade d3 to the latest 5.x in the brouter-web, so we could integrate the Heightgraph as it; an integration plugin (similar to the [Elevation integration plugin](https://github.com/nrenner/brouter-web/blob/master/js/plugin/Elevation.js)) would be required, but I have written it already - it is simpler than the Elevation integration, thanks to the resize support in Heightgraph, as well as other APIs which I have already integrated into the project to make it easier to embed.
This was my first choice and I took a stab at it, but it got pretty hairy and I fell back to the other option (backporting the Heightgraph to d3 3.x). FWIW this would be the cleanest option, and @nrenner suggested that I should start a discussion on this, so here we go...

On a side note, here's a screenshot of a route in brouter-web with the Heightgraph integrated and enabled:
![screenshot](https://user-images.githubusercontent.com/479131/96327306-b69bcc80-0ff5-11eb-9aa5-e691adc1faf9.png)

**Comment by nrenner** [comment]
The reason I wanted to discuss is to see if there is a way forward with option 2, as I am planning to add ES6 syntax support through Babel anyway - but no modules yet. 

Option 3 is to use the already transpiled and compiled bundle, see PR #345. Probably the preferred option right now.

**Comment by alexcojocaru** [comment]
Forgot to mention option #3 in the initial comment, thanks for doing that. I went with option #3 in the end, because #1 is an ugly hack, and #2 is not straightforward (at least for me).
Since for #3 I had to bring in the whole transpiled Heightgraph (i.e. Heightgraph + d3 + many other dependencies), that means some code duplication in the brouter-web bundle, which makes it bigger than it should be.
